src/imitation/rewards/reward_nets.py

```diff
@@ -318,6 +302,210 @@ def forward(self, state, action, next_state, done):
         return outputs
 
 
+class NormalizedRewardNet(RewardNetWrapper):
+    """A reward net that normalizes the output of its base net.
+
+    Only requires the implementation of a forward pass (calculating rewards given
+    a batch of states, actions, next states and dones) as it inherits RewardNet.
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        base: RewardNet,
+        rew_normalize_class: Type[nn.Module],
+        normalize_images: bool = True,
+    ):
+        """Initialize the NormalizedRewardNet.
+
+        Args:
+            observation_space: the observation space of the environment
+            action_space: the action space of the environment
+            base: a base RewardNet
+            rew_normalize_class: The class to use to normalize rewards. This
+                can be any nn.Module that preserves the shape; e.g. `nn.Identity`,
+                `nn.BatchNorm*`, `nn.LayerNorm`, or `networks.RunningNorm`.
+            normalize_images: passed through to `RewardNet.__init__`,
+                see its documentation
+        """
+        super().__init__(
+            observation_space=observation_space,
+            action_space=action_space,
+            base=base,
+            normalize_images=normalize_images,
+        )
+        # assuming reward is always a scalar
+        self.rew_normalize_layer = (
+            rew_normalize_class(1) if rew_normalize_class else None
+        )
+
+    def predict_processed(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Compute normalized rewards for a batch of transitions without gradients.
+
+        Args:
+            state: Current states of shape `(batch_size,) + state_shape`.
+            action: Actions of shape `(batch_size,) + action_shape`.
+            next_state: Successor states of shape `(batch_size,) + state_shape`.
+            done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
+
+        Returns:
+            Computed normalized rewards of shape `(batch_size,`).
+        """
+        rew = self.base.predict(state, action, next_state, done)
+        rew_th = th.as_tensor(rew, device=self.device)
+        rew = self.rew_normalize_layer(rew_th).detach().cpu().numpy().flatten()
+        assert rew.shape == state.shape[:1]
+        return rew
+
+    def forward(
+        self,
+        state: th.Tensor,
+        action: th.Tensor,
+        next_state: th.Tensor,
+        done: th.Tensor,
+    ):
+        return self.base(state, action, next_state, done)
```

Comment:
Shouldn't this be?
```suggestion
        return self.base.forward(state, action, next_state, done)
```
(Maybe it's the same due to PyTorch magic...?)

Reply:
Yeah. It is [PyTorch magic](https://github.com/pytorch/pytorch/blob/ad88816c864c04b478009eec3964d9210291a7a7/torch/nn/modules/module.py#L1175). It was [suggested](https://stackoverflow.com/questions/58508190/in-pytorch-what-is-the-difference-between-forward-and-an-ordinary-method) not to call `forward()`, but to call the layer itself.